src/main/scala/nl.knaw.dans.easy.deposit/docs/DatasetXml.scala

```diff
@@ -0,0 +1,132 @@
+/**
+ * Copyright (C) 2018 DANS - Data Archiving and Networked Services (info@dans.knaw.nl)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package nl.knaw.dans.easy.deposit.docs
+
+import nl.knaw.dans.easy.deposit.docs.DatasetMetadata.DateQualifier.{ available, created }
+import nl.knaw.dans.easy.deposit.docs.DatasetMetadata._
+
+import scala.util.Try
+import scala.xml.{ Attribute, Elem, Null, PrefixedAttribute }
+
+object DatasetXml {
+  private type MaybeAttribute = Option[Attribute]
+
+  private val otherDates = DateQualifier.values.filter(d => d != created && d != available).toSeq
+  private val qualifier = "qualifier"
+
+  def apply(dm: DatasetMetadata): Try[Elem] = Try {
+    val lang = dm.languageOfDescription.map(l => new PrefixedAttribute("xml", "lang", l.key, Null))
+    <ddm:DDM
+      xmlns:dc="http://purl.org/dc/elements/1.1/"
+      xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+      xmlns:dcterms="http://purl.org/dc/terms/"
+      xmlns:dcx-dai="http://easy.dans.knaw.nl/schemas/dcx/dai/"
+      xmlns:ddm="http://easy.dans.knaw.nl/schemas/md/ddm/"
+      xsi:schemaLocation="http://easy.dans.knaw.nl/schemas/md/ddm/ http://easy.dans.knaw.nl/schemas/md/2017/09/ddm.xsd"
+    >
+      <ddm:profile>
+        { requiredElems(dm.titles, "dcterms:title").addAttr(lang) }
+        { elems(dm.alternativeTitles, "dcterms:alternative").addAttr(lang) }
+        { requiredElems(dm.descriptions, "dc:description").addAttr(lang) }
+        { requiredElems(dm.creators, "dcx-dai:creatorDetails", lang) }
+        { requiredElems(dm.dates.map(filter(_, Seq(created))), "ddm:created") }
+        { requiredElems(dm.dates.map(filter(_, Seq(available))), "ddm:available") }
+        { requiredElems(dm.audiences, "ddm:audience") }
+        { requiredElems(dm.accessRights.map(Seq(_)), "ddm:accessRights") }
+      </ddm:profile>
+      <ddm:dcmiMetadata>
+        { elems(dm.contributors, "dcx-dai:creatorDetails", lang) }
+        { elems(dm.publishers, "dcterms:publisher").addAttr(lang) }
+        { elems(dm.sources, "dc:source").addAttr(lang) }
+        { elems(dm.dates.map(filter(_, otherDates)), qualifier) }
+        { optionalElem(dm.license, "dcterms:license") /* TODO xsi:type="dcterms:URI" not supported by json */ }
```

Comment:
I assume you will continue this? I'm missing some fields...

General remark about this `dataset.xml`:
Have you looked at [AddDatasetMetadataToDeposit.scala](https://github.com/rvanheest-DANS-KNAW/easy-split-multi-deposit/blob/ea7c2dc3d6a78cacd779d2efcb2d799e399e22cf/src/main/scala/nl.knaw.dans.easy.multideposit/actions/AddDatasetMetadataToDeposit.scala#L279-L278) in easy-split-multi-deposit. Not about the coding style (I like this one way more!), but about the content. Can we make sure that both applications produce similar DDM with the same tags, in the same order, etc.?

For example: you're using `dcterms:titles` while SMD is using `dc:titles`; you're using `dc:description` while SMD is using `dcterms:description`, etc.